modules/cli/src/Main.scala

```diff
@@ -15,26 +15,56 @@
 
 package smithytranslate.cli
 
-import smithytranslate.cli.opts.OpenAPIJsonSchemaOpts
-import smithytranslate.cli.runners.OpenApi
-import smithytranslate.cli.opts.ProtoOpts
-import smithytranslate.cli.runners.Proto
+import cats.data.Validated
+import com.monovore.decline.Opts
+import smithytranslate.cli.opts.FormatterOpts.FormatOpts
+import smithytranslate.cli.opts.SmithyTranslateCommand.{
+  Format,
+  OpenApiTranslate,
+  ProtoTranslate
+}
+import smithytranslate.cli.opts.{
+  FormatterOpts,
+  OpenAPIJsonSchemaOpts,
+  ProtoOpts,
+  SmithyTranslateCommand
+}
+import smithytranslate.cli.runners.formatter.Formatter.reformat
+import smithytranslate.cli.runners.formatter.FormatterError
+import smithytranslate.cli.runners.{OpenApi, Proto}
 
 object Main
     extends smithytranslate.cli.CommandApp(
       name = "smithy-translate",
       header =
-        "Provides conversion commands to and from Smithy to other languages.",
+        "utils to convert  To and From Smithy to other Languages and to format Smithy files.",
       main = {
-        val cli = OpenAPIJsonSchemaOpts.openApiToSmithy
-          .orElse(ProtoOpts.smithyToProto)
-          .orElse(OpenAPIJsonSchemaOpts.jsonSchemaToSmithy)
+        val cli: Opts[SmithyTranslateCommand] =
+          OpenAPIJsonSchemaOpts.openApiToSmithy
+            .orElse(ProtoOpts.smithyToProto)
+            .orElse(OpenAPIJsonSchemaOpts.jsonSchemaToSmithy)
+            .orElse(FormatterOpts.format)
         cli map {
-          case opts: OpenAPIJsonSchemaOpts if opts.isOpenapi =>
+          case OpenApiTranslate(opts) if opts.isOpenapi =>
             OpenApi.runOpenApi(opts)
-          case opts: OpenAPIJsonSchemaOpts if !opts.isOpenapi =>
-            OpenApi.runJsonSchema(opts)
-          case opts: ProtoOpts => Proto.runFromCli(opts)
+          case OpenApiTranslate(opts) => OpenApi.runJsonSchema(opts)
+          case ProtoTranslate(opts)   => Proto.runFromCli(opts)
+          case Format(FormatOpts(files, noClobber)) =>
+            files.toList.flatMap(file => reformat(file, noClobber)).foreach {
+              case Validated.Valid(smithyFile) =>
+                println(s"Reformatted $smithyFile")
+              case Validated.Invalid(formatterError) =>
```

Comment:
minor: you can probably extract the `FormatterError` directly with:
```scala
case Validated.Invalid(FormatterError.UnableToParse(message)) =>
  ...
case Validated.Invalid(FormatterError.InvalidModel(message)) =>
  ...
```

Reply:
i actually prefer this, i think it delineates error and success states better